src/crimson/os/seastore/onode_manager/staged-fltree/node.h

```diff
@@ -79,38 +80,72 @@ class tree_cursor_t final
   bool is_end() const { return position.is_end(); }
 
   /// Returns the key view in tree if it is not an end cursor.
-  const key_view_t& get_key_view() const;
+  const key_view_t& get_key_view() const {
+    maybe_update_cache();
+    return cache.get_key_view();
+  }
 
   /// Returns the value pointer in tree if it is not an end cursor.
-  const onode_t* get_p_value() const;
+  const onode_t* get_p_value() const {
+    maybe_update_cache();
+    return cache.get_p_value();
+  }
 
  private:
   tree_cursor_t(Ref<LeafNode>, const search_position_t&);
   tree_cursor_t(Ref<LeafNode>, const search_position_t&,
-                const key_view_t& key, const onode_t*, layout_version_t);
+                const key_view_t&, const onode_t*);
   // lookup reaches the end, contain leaf node for further insert
   tree_cursor_t(Ref<LeafNode>);
+
   const search_position_t& get_position() const { return position; }
-  Ref<LeafNode> get_leaf_node() { return leaf_node; }
+  Ref<LeafNode> get_leaf_node() { return ref_node; }
   template <bool VALIDATE>
   void update_track(Ref<LeafNode>, const search_position_t&);
-  void update_kv(const key_view_t&, const onode_t*, layout_version_t) const;
-  void ensure_kv() const;
+  void update_cache_fast(LeafNode&, const key_view_t&, const onode_t*) const;
+  void maybe_update_cache() const;
 
- private:
   /**
    * Reversed resource management (tree_cursor_t)
    *
    * tree_cursor_t holds a reference to the LeafNode, so the LeafNode will be
    * alive as long as any of it's cursors is still referenced by user.
    */
-  Ref<LeafNode> leaf_node;
+  Ref<LeafNode> ref_node;
```

Comment:
FWIW, i like `leaf_node` better as it's more specific in this context. but anyway ...

Reply:
renamed to `ref_leaf_node`.